### ISSUE TITLE
How to reproduce the same result as in the paper

### ISSUE THREAD

**Issue by ali2iptoki** [NVIDIA/NeMo#2622]
I tried to reproduce the result of speakerrecognition_speakernet results which:

` description="SpeakerNet_recognition model trained end-to-end for speaker recognition purposes with cross_entropy loss. It was trained on voxceleb 1, voxceleb 2 dev datasets and augmented with musan music and noise. Speaker Recognition model achieves 2.65% EER on voxceleb-O cleaned trial file",`

So, to infer label we need two files:
1- train_manifest=/path/to/train/manifest/file'
2- test_manifest=/path/to/train/manifest/file'

and then call the provided script: `python speaker_reco_infer.py --spkr_model='/path/to/.nemo/file' --train_manifest=/path/to/train/manifest/file --test_manifest=/path/to/train/manifest/file --batch_size=32`

So, correct me please. The test_manifest is `https://www.robots.ox.ac.uk/~vgg/data/voxceleb/meta/veri_test2.txt`. But Usualy we need it in json format where can I find this file?

Where can I find the `train_manifest`?

If I am wrong, how can reproduce the speaker recognition on voxceleb1 and voxceleb2 as in the paper? Indeed, you was using Voxceleb1 and Voxceleb2 for training. Can for example, we use Voxceleb1 for training and then Voxceleb2 for testing?

**Comment by nithinraok** [comment]
Download voxceleb1 and voxceleb2 dataset from https://www.robots.ox.ac.uk/~vgg/data/voxceleb/vox1.html. use dev part of both datasets to create train and validation sets using scripts provided in https://github.com/NVIDIA/NeMo/blob/main/scripts/speaker_recognition/scp_to_manifest.py following instructions from https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/main/asr/speaker_recognition/datasets.html#all-other-datasets. 


As said in previous issues and also by email, speaker verification, recogniton and diarization are different speaker tasks. In general we develop a speaker embedding model using speaker_reco.py file and use the pretrained embeddings for speaker verification tasks. The number reported in paper can be reproduced using the pretrained models provided in [NGC catalog ](https://ngc.nvidia.com/catalog/models/nvidia:nemo:speakerverification_speakernet)and with the manifest file generated from https://www.robots.ox.ac.uk/~vgg/data/voxceleb/meta/veri_test2.txt. Manifest json file is very easy to generate which contains rows with keys as audio_filepath, duration and label. So create a simple script to write corresponding manifest files using ids from voxceleb trial file. You would need certain background idea on these tasks else it would be difficult to follow. 

Sample row from manifest file I created:
`{"audio_filepath": "/data/voxceleb1/test_wav/id10274/wH6ebv-K8AA/00004.wav", "duration": 7.2800625, "label": "id10274"}` 

Regarding the exact manifest files, if it's very unclear to follow the instructions let me know I will create a script to generate train and dev manifest files for speaker embedding training with voxceleb datasets.

**Comment by haha010508** [comment]
i get the eer=0.8, this is my code 
`#!/bin/sh
CUDA_VISIBLE_DEVICES=3 python extract_speaker_embeddings.py --manifest=voxceleb1_test_manifest.json --model_path='./speakerverification_en_titanet_large/model' --embedding_dir='./speakerverification_en_titanet_large/embedding'

CUDA_VISIBLE_DEVICES=3 python voxceleb_eval.py --trial_file='./speech_database/voxceleb/vox_trial_pairs/veri_test.txt' --emb='./speakerverification_en_titanet_large/embedding/embeddings/voxceleb1_test_manifest_embeddings.pkl' `

and i found the model is:
`Found existing object .cache/torch/NeMo/NeMo_1.17.0/titanet-l/11ba0924fdf87c049e339adbf6899d48/titanet-l.nemo.`
`100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 37720/37720 [00:00<00:00, 110401.53it/s]
0.80
`
So I want to know what went wrong? thanks very much!!

**Comment by nithinraok** [comment]
Hi, what you have performed is right, but our reported results are based on [`veri_test2.txt`](https://www.robots.ox.ac.uk/~vgg/data/voxceleb/meta/veri_test2.txt) not [veri_test.txt](https://www.robots.ox.ac.uk/~vgg/data/voxceleb/meta/veri_test.txt)

**Comment by ukemamaster** [comment]
@haha010508 How does your `voxceleb1_test_manifest.json` look like?
